femtools/Particles.F90

```diff
@@ -755,9 +875,10 @@ subroutine read_particles_from_file(n_particles, subgroup_name, subgroup_path, &
       call read_attrs(h5_id, dim, attr_counts%old_fields, old_field_names, old_field_vals, prefix="old%")
 
       ! don't use a global check for this particle
-      call create_single_particle(p_list, xfield, &
-           positions, id(1), proc_id(1), trim(particle_name), dim, &
-           attr_counts, attr_vals, old_attr_vals, old_field_vals, global=.false.)
+      call create_single_particle(p_list, list_id, xfield, &
+           positions, id(1), proc_id(1), dim, &
+           attr_counts, attr_vals, old_attr_vals, old_field_vals, global=.false., particle_number=particle_number)
```

Comment:
Can you use `particle_number` as the loop counter (or `i` as the particle number parameter)? Seems redundant to have both

Reply:
Seems the whole `particle_number` parameter was redundant anyway